system_metrics_collector/launch/system_cpu_and_memory.launch.py

```diff
@@ -12,24 +12,68 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Launch system CPU and system memory metrics collector nodes."""
-
+import os
+import yaml
 
+from ament_index_python.packages import get_package_share_directory
 from launch import LaunchDescription
+from launch.actions import DeclareLaunchArgument
+from launch.substitutions import LaunchConfiguration
 from launch_ros.actions import Node
 
+# Argument names
+CPU_NODE_NAME = 'linux_cpu_collector_node_name'
+MEMORY_NODE_NAME = 'linux_memory_collector_node_name'
+MEASUREMENT_PERIOD = 'measurement_period'
+PUBLISH_PERIOD = 'publish_period'
+PUBLISH_TOPIC = 'publish_topic'
 
+"""Launch system CPU and system memory metrics collector nodes."""
 def generate_launch_description():
+    # Default to included config file
+    config_filepath = os.path.join(
+        get_package_share_directory('system_metrics_collector'),
+        'config',
+        'system_metrics_collector_config.yaml')
+    with open(config_filepath, 'r') as f:
+        config_file_contents = f.read()
+    config_yaml_data = yaml.safe_load(config_file_contents)
```

Comment:
Can't this throw if the config contains invalid yaml?

Reply:
So what do you want the behavior of an invalid YAML to be?

Reply:
The reading of a YAML file has been removed altogether.